.github/workflows/pr-workflow.yml

```diff
@@ -0,0 +1,48 @@
+name: PR Workflow
+run-name: ${{ github.actor }} pull request
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types:
+      - opened
+    branches:
+      - main
+
+jobs:
+  test_and_lint_client:
+    name: Test and lint client
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./client
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '18'
+      - name: Install dependencies
+        run: npm install
```

Comment:
Should this be npm ci or npm install?

Reply:
It can be changed to `npm ci` in the PR which pushes `package-lock`.